### ISSUE TITLE
nan 2.4+ causes fatal error

### ISSUE THREAD

**Issue by jehy** [nodejs/nan#798]
Hi! Recently nan, as a transitive dependency of modern-syslog, was updated from 2.3.5 to 2.10.

After it our application could not be started any more due to this error:

```
# Fatal error in ../deps/v8/src/objects.cc, line 3577
# Check failed: receiver->IsJSFunction().
```
I checked both with latest version of Node 8 and Node 10 - error is same.
May be line numbers are different (3577 is for node 8).

Then I began downgrading nan.
2.9 - error still here
2.8 - error still here
2.7 - error still here
2.6 - error still here
2.5 - error still here
2.4 - error still here
2.3.5 - everything fine!

Seems like something happened in 2.4.0 and it broke our use case.
Unfortunately, I am still unable to create a reproducable sample of this bug, but could you take a look at this?

**Comment by jehy** [comment]
Update: I updated modern-syslog to latest version and error is gone.

However, I still suppose that there can be a kind of bug in nan - at least it could give a clear error message...

That's the fix in modern-syslog which made nan@2.10 great again for me: https://github.com/strongloop/modern-syslog/pull/17

**Comment by bnoordhuis** [comment]
Since it was a bug in modern-syslog, I'll go ahead and close this out. The assert is a sanity check in V8 that was doing its work.
